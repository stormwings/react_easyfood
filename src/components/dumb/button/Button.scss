@import './../../../styles/variables';
@import './../../../styles/fonts';
@import './../../../styles/mixins';

#button {
  .button {
    padding: 4px 12px;
    margin: 0 auto;

    &.button-success {
      border-radius: 25px;
      background-color: $colorPink;
      color: $colorWhite;
    }

    &.button-empty {
      color: $colorPink;
    }

    &.button-success-outline {
      color: $colorPink;
      border: 2px solid $colorPink;
      border-radius: 25px;
    }
  }

  .image-button {
    width: 20px;
    padding: 10px;
    border-radius: 25px;
    background-color: $colorPink;
    color: $colorWhite;
    @include flex_parent_center_height();
  }
}