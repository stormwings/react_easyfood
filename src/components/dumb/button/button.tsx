import React, { FC } from 'react';
import styled from 'styled-components';

import { useTranslation, Trans } from 'react-i18next';

export interface ButtonProps {
  /**
   * Text of the button
   *
   * @default ""
   **/
  content: string;
  /**
   * function to execute on click
   */
  onClick?: (e: unknown) => void;
}
export const FancyButton = styled.button`
  border: 1px solid #eee;
  border-radius: 3px;
  background-color: white;
  cursor: pointer;
  font-size: 15px;
  padding: 3px 10px;
  margin: 10px;
`;

export const Button: FC<ButtonProps> = props => {
  // you can use hook for the translation
  // https://react.i18next.com/latest/usetranslation-hook
  const { t } = useTranslation();
  return (
    <>
      <div>
        {/* Or you can use the Trans component 
            https://react.i18next.com/latest/trans-component */}
        <Trans i18nKey="title" />
      </div>
      <FancyButton onClick={props.onClick}>{props.content}</FancyButton>
      <div>{t('subTitle')}</div>
    </>
  );
};
export default Button;
