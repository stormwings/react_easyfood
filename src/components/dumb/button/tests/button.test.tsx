import renderer from 'react-test-renderer';
import React from 'react';

import { Button } from '../button';

describe('Button', () => {
  it('Renders with text', () => {
    const tree = renderer.create(<Button content="My Button" />);
    expect(tree).toMatchSnapshot();
  });
});
