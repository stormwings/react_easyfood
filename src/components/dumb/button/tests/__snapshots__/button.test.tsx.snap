// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Renders with text 1`] = `
Array [
  <div />,
  <button
    className="sc-bdVaJa jyKrJs"
  >
    My Button
  </button>,
  <div>
    subTitle
  </div>,
]
`;
